# Excessive white-space on right-hand side in icon view
# When using at allocation->width, take zoom into account
# Launchpad: https://bugs.launchpad.net/ubuntu/+source/nautilus/+bug/362771
# Bugzilla: http://bugzilla.gnome.org/show_bug.cgi?id=579086

Index: nautilus-2.26.2/libnautilus-private/nautilus-icon-container.c
===================================================================
--- nautilus-2.26.2.orig/libnautilus-private/nautilus-icon-container.c	2009-04-20 23:31:26.000000000 +0100
+++ nautilus-2.26.2/libnautilus-private/nautilus-icon-container.c	2009-04-20 23:31:36.000000000 +0100
@@ -1033,9 +1033,9 @@
 	GtkAllocation *allocation;
 	gboolean reset_scroll_region;
 
-	if (nautilus_icon_container_get_is_fixed_size (container)) {
-		pixels_per_unit = EEL_CANVAS (container)->pixels_per_unit;
+	pixels_per_unit = EEL_CANVAS (container)->pixels_per_unit;
 
+	if (nautilus_icon_container_get_is_fixed_size (container)) {
 		/* Set the scroll region to the size of the container allocation */
 		allocation = &GTK_WIDGET (container)->allocation;
 		eel_canvas_set_scroll_region
@@ -1090,7 +1090,7 @@
 	if (nautilus_icon_container_is_auto_layout (container)) {
 		allocation = &GTK_WIDGET (container)->allocation;
 		x1 = MIN (x1, 0);
-		x2 = MAX (x2, allocation->width);
+		x2 = MAX (x2, allocation->width / pixels_per_unit);
 		y1 = 0;
 	} else {
 		/* Otherwise we add the padding that is at the start of the
